src/others/xml/parsers/lv_xml_tabview_parser.c

```diff
@@ -87,6 +87,33 @@ void lv_xml_tabview_tab_apply(lv_xml_parser_state_t * state, const char ** attrs
     lv_xml_obj_apply(state, attrs);
 }
 
+void * lv_xml_tabview_tab_button_create(lv_xml_parser_state_t * state, const char ** attrs)
+{
+    lv_obj_t * tv = lv_xml_state_get_parent(state);
+    int32_t btn_cnt = lv_tabview_get_tab_count(tv);
+    if(btn_cnt == 0) {
+        LV_LOG_WARN("There are no buttons on the tab view. Get tab buttons when the tabs are already created");
+        return NULL;
+    }
+
+    const char * index_str = lv_xml_get_value_of(attrs, "index");
+    int32_t index_int = index_str ? lv_xml_atoi(index_str) : 0;
+    if(LV_ABS(index_int) >= btn_cnt) {
```

Comment:
If there are 2 btns, -2 would not be out of range but would still trigger this error condition.

```suggestion
    if((index_int < 0 ? -index_int - 1 : index_int) >= btn_cnt) {
```

(or `index_int < 0 ? ~index_int : index_int`)